porch/controllers/rootsyncrollout/api/v1alpha1/rootsyncrollout_types.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RootSyncRolloutSpec
+type RootSyncRolloutSpec struct {
+	Targets []ClusterTargetSelector `json:"targets,omitempty"`
+
+	Packages PackageSelector `json:"packages,omitempty"`
+}
+
+type ClusterTargetSelector struct {
+	ClusterType ClusterType `json:"clusterType,omitempty"`
```

Comment:
Maybe this should be a GVK or GK?  We can see how it goes though...

Reply:
Yeah, we could make this a GK/GVK. That will help with discovery for clusters that are represented as resources in the local cluster, but we still need logic to know how to work with those clusters, i.e. connecting to them. Also, there might be situations where we want to discover clusters through some other mean that listing resources, although we could argue that all APIs must reflect clusters as resources.